chainer/dataset/download.py

```diff
@@ -0,0 +1,148 @@
+from __future__ import print_function
+import hashlib
+import os
+import shutil
+import tempfile
+
+import filelock
+from six.moves.urllib import request
+
+
+_dataset_root = os.environ.get('CHAINER_DATASET_ROOT',
+                               os.path.expanduser('~/.chainer/dataset'))
+
+
+def get_dataset_root():
+    """Gets the path to the root directory to download and cache datasets.
+
+    Returns:
+        str: The path to the dataset root directory.
+
+    """
+    return _dataset_root
+
+
+def set_dataset_root(path):
+    """Sets the root directory to download and cache datasets.
+
+    There are two ways to set the dataset root directory. One is by setting the
+    environmant variable ``CHAINER_DATASET_ROOT``. The other is by using this
+    function. The default dataset root is ``$HOME/.chainer/dataset``.
```

Comment:
What happens when we change the root directory by `set_dataset_root` and change `CHAINER_DATASET_ROOT` after that? I guess the second manipulation does not have any effect because we check the value of `CHAINER_DATASET_ROOT` once.